addons/osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsGitCredentials.java

```diff
@@ -0,0 +1,96 @@
+package io.fabric8.launcher.osio.jenkins;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+import javax.ws.rs.core.HttpHeaders;
+
+import io.fabric8.launcher.base.http.ExternalRequest;
+import io.fabric8.launcher.base.identity.Identity;
+import io.fabric8.launcher.base.identity.IdentityVisitor;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.core.spi.IdentityProvider;
+import io.fabric8.launcher.osio.OsioConfigs;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+
+import static io.fabric8.utils.URLUtils.pathJoin;
+
+@RequestScoped
+public class JenkinsGitCredentials {
+    public JenkinsGitCredentials() {
+    }
+
+    JenkinsGitCredentials(String authorization, String gitToken) {
```

Comment:
I assume this constructor is using for tests only. I think it should be better if it accepted an `IdentityProvider`?